Tools/deploy-mac.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import errno
+import os
+import re
+import shutil
+import subprocess
+
+qtPath = None
+verbose = False
+
+def splitPath(path):
+	folders = []
+	while True:
+		path, folder = os.path.split(path)
+		if folder != '':
+			folders.append(folder)
+		else:
+			if path != '':
+				folders.append(path)
+			break
+	folders.reverse()
+	return folders
+
+def joinPath(path):
+	return reduce(os.path.join, path, '')
+
+def findFramework(path):
+	child = []
+	while path and not path[-1].endswith('.framework'):
+		child.append(path.pop())
+	child.reverse()
+	return path, child
+
+def findQtPath(path):
+	parent, child = findFramework(splitPath(path))
+	return joinPath(parent[:-2])
+
+def makedirs(path):
+	split = splitPath(path)
+	accum = []
+	split.reverse()
+	while split:
+		accum.append(split.pop())
+		newPath = joinPath(accum)
+		if newPath == '/':
+			continue
+		try:
+			os.mkdir(newPath)
+		except OSError as e:
+			if e.errno != errno.EEXIST:
+				raise
+
+
+def parseOtoolLine(line, execPath, root):
+	if not line.startswith('\t'):
+		return None, None, None, None
+	line = line[1:]
+	match = re.match('([@/].*) \(compatibility version.*\)', line)
+	path = match.group(1)
+	split = splitPath(path)
+	newExecPath = ['@executable_path', '..', 'Frameworks']
+	newPath = execPath[:-1]
+	newPath.append('Frameworks')
+	if split[:3] == ['/', 'usr', 'lib'] or split[:2] == ['/', 'System']:
+		return None, None, None, None
+	if split[0] == '@executable_path':
+		split[:1] = execPath
+	if split[0] == '/' and not os.access(joinPath(split), os.F_OK):
+		split[:1] = root
```

Comment:
Not sure which lines you're talking about here. parseOtoolLine is for figuring out which libraries and frameworks a mach-o pulls in, then making a new path relative to the executable for where they will live. Line 66-67 are to keep it from pulling in system libraries and frameworks. 68-69 is for includes that are already relative to the exec path, and lines 70-71 have to do with using a buildroot instead of on a native system (which is relevant for cross-compiling).